ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh

```diff
@@ -248,3 +289,37 @@ if ! wait "$!"; then
   # TODO: gather logs??
   exit 1
 fi
+
+if [ "${ADDITIONAL_WORKERS}" == "0" ]; then
+    echo "No additional workers requested"
+    exit 0
+fi
+
+oc extract -n openshift-machine-api secret/worker-user-data-managed --keys=userData --to=- > "${SHARED_DIR}"/worker.ign
+
+echo -e "\nCopying ignition files into bastion host..."
+chmod 644 "${SHARED_DIR}"/*.ign
+scp "${SSHOPTS[@]}" "${SHARED_DIR}"/*.ign "root@${AUX_HOST}:/opt/html/${CLUSTER_NAME}/"
+
+for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
```

Comment:
Should this logic go into the upi/baremetal/heterogeneous step directly? All the scale-out logic is expected there and you'd need less stuff to do there for it.

Reply:
Thanks @aleskandro , I will update. Please move to https://github.com/openshift/release/pull/47637 for the updates later.